frontend/src/app/services/events.ts

```diff
@@ -76,26 +79,30 @@ export class EventHandler {
       ...currentState,
       date: dayState.date,
       stats: dayState.stats,
+      mitigations,
       eventMitigations,
     };
 
     updateEventData(eventInput);
 
     this.eventStateHistory[currentDate] = currentState;
 
-    const active = shuffle(currentState.triggerStates.filter(ts =>
+    const active = currentState.triggerStates.filter(ts =>
       ts.activeBefore === undefined
-        || ts.trigger.reactivateAfter !== undefined && ts.activeBefore >= ts.trigger.reactivateAfter));
-    const triggerState = active.find(ts => ts.trigger.condition(eventInput));
+        || ts.trigger.reactivateAfter !== undefined && ts.activeBefore >= ts.trigger.reactivateAfter);
+    const triggered = active.filter(ts => ts.trigger.condition(eventInput));
 
-    if (!triggerState) return;
+    if (!triggered) return;
```

Comment:
Always false since `![] === false`, check for `.length` if empty array makes a difference.

Reply:
My bad. After several mistakes like this, I am finally learning.